src/libs/antares/utils/utils.cpp

```diff
@@ -172,5 +172,30 @@ double floorDiv(double numerator, double denominator)
     return std::floor(std::round(numerator / denominator * largeValue) / largeValue);
 }
 
+bool checkAllElementsIdenticalOrOne(std::vector<unsigned> w)
+{
+    auto first_one = std::remove(w.begin(), w.end(), 1); // Reject all 1 to the end
+    return std::adjacent_find(w.begin(), first_one, std::not_equal_to<unsigned>()) == first_one;
+}
+
+std::pair<bool, std::string> checkAllElementsIdenticalOrOne(
+  std::vector<std::pair<unsigned, std::string>>& p)
+{
+    // Reject all 1 to the end
+    auto first_one = std::ranges::remove_if(p, [](const auto& pair) { return pair.first == 1; });
+    auto width = p.begin()->first;
+    for (auto pair = p.begin(); pair < first_one.begin(); ++pair)
+    {
+        if (pair->first != width)
+        {
+            logs.error() << "Inconsitent time series width, found: " << pair->first
+                         << " Previous was " << width << " for " << pair->second;
+
+            return {false, pair->second};
+        }
+    }
+    return {true, ""};
+}
```

Comment:
```suggestion
std::pair<bool, std::string> checkAllElementsIdenticalOrOne(
  std::vector<std::pair<unsigned, std::string>>& p)
{
    // Reject all 1 to the end
    auto first_one = std::ranges::remove_if(p, [](const auto& pair) { return pair.first == 1; });
    auto width = p.begin()->first;
    for (const auto& [w, label] : p)
    {
        if (w != width)
        {
            logs.error() << "Inconsitent time series width, found: " << w
                         << " Previous was " << width << " for " << label;

            return false;
        }
    }
    return true;
}
```
The return type is not necessary